cmd/medius/images/push.go

```diff
@@ -107,45 +108,33 @@ func NewPublishImagesCommand(options *common.Options) *cobra.Command {
 }
 
 func (b *buildAndPublish) Do(entry *common.Entry, timestamp time.Time) ([]string, error) {
-	metadata := entry.Artifact.Metadata()
-	artifactInfo, err := entry.Artifact.Inspect()
+	metadata := entry.Artifacts[0].Metadata()
+	artifactInfo, err := entry.Artifacts[0].Inspect()
 	if err != nil {
 		return nil, fmt.Errorf("error introspecting artifact %q: %v", metadata.Describe(), err)
 	}
-	b.Log.Infof("Remote artifact checksum: %q", artifactInfo.SHA256Sum)
 
-	imageSha, err := b.getImageSha(metadata.Describe())
+	images, artifacts, err := b.buildImages(entry)
 	if err != nil {
 		return nil, err
 	}
-	if imageSha == artifactInfo.SHA256Sum && !b.Options.PublishImagesOptions.ForceBuild {
-		b.Log.Info("Nothing to do.")
-		return nil, nil
-	}
-	if errors.Is(b.Ctx.Err(), context.Canceled) {
-		return nil, b.Ctx.Err()
-	}
-
-	b.Log.Infof("Rebuild needed, downloading %q ...", artifactInfo.DownloadURL)
-	file, err := b.getArtifact(artifactInfo)
-	if err != nil {
-		return nil, err
-	}
-	defer os.Remove(file)
+	defer cleanupArtifacts(artifacts)
 
-	b.Log.Info("Building containerdisk ...")
-	containerDisk, err := build.ContainerDisk(file, build.ContainerDiskConfig(artifactInfo.SHA256Sum, metadata.AdditionalLabels))
+	containerDiskIndex, err := build.ContainerDiskIndex(images)
 	if err != nil {
-		return nil, fmt.Errorf("error creating the containerdisk : %v", err)
-	}
-	if errors.Is(b.Ctx.Err(), context.Canceled) {
-		return nil, b.Ctx.Err()
+		return nil, fmt.Errorf("error creating the containerdisk index : %v", err)
 	}
 
 	names := prepareTags(timestamp, b.Options.PublishImagesOptions.TargetRegistry, entry, artifactInfo)
 	for _, name := range names {
-		if err := b.pushImage(containerDisk, name); err != nil {
-			return nil, err
+		if containerDiskIndex != nil {
```

Comment:
Can you make this dependent on `len(images)`? If len is not greater than 1 we know we do not require an index.